core/unix/module_elf.c

```diff
@@ -1688,8 +1688,15 @@ module_init_rseq(module_area_t *ma, bool at_map)
                     goto module_init_rseq_cleanup;
                 /* We assume this is a full mapping and it's safe to read the data
                  * (a partial map shouldn't make it to module list processing).
+                 * We do perform a sanity check to handle unusual non-relocated
+                 * cases (it's possible this array is not in a loaded segment?).
                  */
-                rseq_process_entry((struct rseq_cs *)(*ptrs + load_offs), entry_offs);
+                if (*ptrs < ma->start || *ptrs > ma->end) {
+                    SYSLOG_INTERNAL_WARNING(RSEQ_PTR_ARRAY_SEC_NAME
+                                            " is not in memory. Aborting rseq parsing.");
+                    goto module_init_rseq_cleanup;
+                }
+                rseq_process_entry((struct rseq_cs *)(*ptrs), entry_offs);
```

Comment:
There is a comment a few lines above, "[..] only need the offset if relocations have not yet been applied [..]" Does this need to be updated and is obsolete?

Reply:
No, but *ptrs needs +entry_offset.  Fixed.  Tested by making rseq eager and running from-init test: no relocs so needs offset.

Reply:
Ok, I don't know enough details but the comment makes sense now.